### PR TITLE
cleaning up documentation of arrayMerge options

### PR DESCRIPTION
**Pull request by CWSites**
You can see the formatted sample here: https://github.com/CWSites/deepmerge

### CODE REVIEW

**Comment by CWSites** [comment]
I don't understand why tests are failing for a readme.md update, in the details it's complaining about a node_module `esprima`

**Comment by CWSites** [comment]
@TehShrike could you help clarify the difference between `overriteMerge` and `dontMerge`? In your examples they seem to be doing the same thing, `x` is wiped out and `y` remains.

**Comment by TehShrike** [comment]
Ah yeah, I don't know why there are two examples there, we only need one of them.

File: readme.md [hunk 1]
```diff
@@ -190,6 +192,15 @@ merge(
 ) // => [{ a: true, b: true }, 'ah yup']
 ```
 
+### `arrayMerge: customFunction`
+
+If you want to supply your own custom method then pass it as an option to `arrayMerge`
+
+```js
+const customFunction = ...
```
Comment (TehShrike) [inline readme.md]: This is why the tests are failing - that's not valid JS syntax.  All the JS code in the readme is tested to make sure the examples are all correct.

File: readme.md [hunk 2]
```diff
@@ -110,6 +110,8 @@ If an element at the same key is present for both `x` and `y`, the value from
 
 Merging creates a new object, so that neither `x` or `y` is modified.
 
+**Note:** by default, arrays are concatenated rather than replacing the array values of `x` with `y`. If you want to change how this works then see `arrayMerge` section below.
```
Comment (TehShrike) [inline readme.md]: Probably better to replace the first sentence with the shorter "By default, arrays are merged by concatenating them."

File: readme.md [hunk 3]
```diff
@@ -128,38 +130,38 @@ merge.all([x, y, z]) // => expected
...
 ## Options
 
-### `arrayMerge`
-deepmerge, by default, concatenates arrays and merges array values. 
-
-There are however nigh-infinite valid ways to merge arrays, and you may want to supply your own method. You can do this by passing an `arrayMerge` function as an option.
-
-The options object will include the default `isMergeableObject` implementation if the top-level consumer didn't pass a custom function in.
-
+### `arrayMerge: overwriteMerge`
```
Comment (TehShrike) [inline readme.md]: This isn't quite right - every section under the Options heading is a different option that you can pass to deepmerge.

Examples of different things you can pass to `arrayMerge` should be fourth level headers, underneath the `arrayMerge` heading.

File: readme.md [hunk 4]
```diff
@@ -190,6 +192,15 @@ merge(
 ) // => [{ a: true, b: true }, 'ah yup']
 ```
 
+### `arrayMerge: customFunction`
```
Comment (TehShrike) [inline readme.md]: This section feels weird to me, since the whole point of this options section is to explain `arrayMerge: yourOwnFunctionHere`

**Comment by CWSites** [comment]
Thanks I'll make the adjustments!

**Comment by CWSites** [comment]
@TehShrike I made your requested changes, please review when you get a moment.

**Pull request merged**